core/lib/libtomcrypt/src/misc/mem_neq.c

```diff
@@ -74,6 +74,11 @@ int mem_neq(const void *a, const void *b, size_t len)
       ++pb;
    }
 
+   ret |= ret >> 4;
+   ret |= ret >> 2;
+   ret |= ret >> 1;
+   ret &= 1;
+
```

Comment:
In my experience GCC compilers aren't quite that smart.  They may optimize the sequence but it'll be data independent which is what you want.